partitioning/gpt/gpt.go

```diff
@@ -589,6 +589,96 @@ func (t *Table) Write() error {
 	return t.syncKernel()
 }
 
+// ReadPartitionData streams the content of a partition (0-indexed) to the writer.
+func (t *Table) ReadPartitionData(partition int, w io.Writer) (int64, error) {
+	if partition < 0 || partition >= len(t.entries) {
+		return 0, fmt.Errorf("partition %d out of range", partition)
+	}
+
+	entry := t.entries[partition]
+	if entry == nil {
+		return 0, fmt.Errorf("partition %d is not allocated", partition)
+	}
+
+	offset := int64(entry.FirstLBA) * int64(t.sectorSize)
+	size := int64(entry.LastLBA-entry.FirstLBA+1) * int64(t.sectorSize)
+
+	section := io.NewSectionReader(t.dev, offset, size)
+
+	return io.Copy(w, section)
+}
+
+// WritePartitionData streams data from the reader into a partition (0-indexed).
+//
+// The input must be exactly the partition size - the method enforces strict size matching:
+//   - Returns error if input is shorter than partition size (partial write)
+//   - Returns error if input is longer than partition size (overflow prevented)
+//
+// This strict validation ensures:
+//   - No uninitialized data remains in the partition
+//   - No data overflows into adjacent partitions
+//   - Clear contract: users must provide partition-sized data
+//
+// To create properly-sized filesystem images:
+//   - VFAT: mkfs.vfat ... image.img && truncate -s <size> image.img
+//   - ext4: mkfs.ext4 ... image.img <size>
+//
+// This behavior matches systemd-repart's Format= directive, which creates
+// filesystems at exactly the partition size.
+func (t *Table) WritePartitionData(partition int, r io.Reader) (int64, error) {
```

Comment:
and probably same here, just `PartitionWriter(partition int) (io.Writer, error)` and use https://pkg.go.dev/io#NewOffsetWriter to implement it. 

we could also return the length of the data, and let the caller handler the rest. 

I don't think GPT implementation here should do too much work here. 

We can in higher level code handle truncating the image/handling overflows, but I think this is too level